File: src/test/java/br/com/controle/financeiro/controller/api/linkbuilder/ClientDTOResourceAssemblerTest.java
package br.com.controle.financeiro.controller.api.linkbuilder;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import java.util.UUID;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.hateoas.EntityModel;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import br.com.controle.financeiro.model.dto.ClientDTO;

@RunWith(SpringRunner.class)
@SpringBootTest(classes = { ClientDTOResourceAssembler.class })
@AutoConfigureMockMvc
@ActiveProfiles(profiles = "test")
public class ClientDTOResourceAssemblerTest {

	@InjectMocks
	private ClientDTOResourceAssembler resourceAssembler;


	@Test
	public void testToResource() {
		ClientDTO clientMock = new ClientDTO();
		clientMock.setId(UUID.randomUUID());
		clientMock.setName("mock");
		EntityModel<ClientDTO> response = resourceAssembler.toModel(clientMock);
		assertTrue(response.hasLinks());
		assertNotNull(response.getLink("self"));
		assertNotNull(response.getLink("clients"));
		assertEquals(clientMock, response.getContent());
	}

}

File: src/test/kotlin/br/com/controle/financeiro/controller/api/linkbuilder/ClientDTOResourceAssemblerTest.kt
package br.com.controle.financeiro.controller.api.linkbuilder

import org.springframework.test.context.ActiveProfiles
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.beans.factory.annotation.Autowired
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import br.com.controle.financeiro.model.dto.ClientDTO
import br.com.controle.financeiro.controller.api.cl

@SpringBootTest
@ActiveProfiles(profiles = ["test"])
class ClientDTOResourceAssemblerTest(
    @Autowired    private val resourceAssembler: ClientDTOResourceAssembler
) {
    @Test
    fun testToResource() {
        val clientMock = ClientDTO("mock", cl.owner, cl.id)
        val response = resourceAssembler.toModel(clientMock)
        Assertions.assertTrue(response.hasLinks())
        Assertions.assertNotNull(response.getLink("self"))
        Assertions.assertNotNull(response.getLink("clients"))
        Assertions.assertEquals(clientMock, response.content)
    }
}
